src/platforms/common/blackpill-f4/README.md

```diff
@@ -81,3 +82,74 @@ mon freq 900k
 ```
 
 For details see the [pull request](https://github.com/blackmagic-debug/blackmagic/pull/783#issue-529197718) that implemented the `mon freq` command.
+
+
+## Performance
+
+System clock is set to either 84 MHz or 96 MHz depending on chosen platform, expecting a 25 MHz HSE crystal.
+
+Because of low expectations to signal integrity or quality wiring, default SWD frequency is reduced to 3 MHz, but in practice 8 MHz is possible, JTAG is slower.
+
+Aux serial runs via USART2 DMA on APB1 (Pclk = 48 MHz) at 732..3000000 baud (6Mbaud with OVER8).
+
+TraceSWO Async capture runs via USART1 DMA on APB2 (Pclk = 96 MHz) at 1465..6000000 baud (12Mbaud with OVER8).
+
+SPI ports are set to Pclk/8 each (use with `bmpflash`). As SPI1 pins may conflict with certain functions, and platform code does not bother restoring them, please soft-reset the probe when done with SPI operations.
+
+## Instructions for Meson buildsystem
+
+### Variant with meson and BMD bootloader enabled (default)
+
+0. Clone the repo and libopencm3 submodule, install toolchains, meson, etc.
+
+```sh
+$ git clone https://github.com/blackmagic-debug/blackmagic.git
+$ cd blackmagic
+$ git submodule update --init
```

Comment:
The style of shell blocks in the documentation, in part to make copying commands into the console easier, is to not include a `$ ` prefix, so please drop that prefix on all shell blocks here and below. Even when the block is for just one command, please use triple-backticks blocks marked for shell on them, this keeps the presentation style consistent.